### PR TITLE
NIFI-10090 Update Guava JRE 28 in NiFi Toolkit to latest version

### PR DESCRIPTION
**Pull request by mr1716**
# Summary

[[NIFI-10090](https://issues.apache.org/jira/browse/NIFI-10090) Updated Guava JRE in the NiFi Toolkit

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [[NIFI-10090](https://issues.apache.org/jira/browse/NIFI-10090) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 8
  - [x] JDK 11
  - [x] JDK 17

### Licensing

- [x] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [x] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [x] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-commons/nifi-property-protection-gcp/pom.xml [hunk 1]
```diff
@@ -53,6 +53,7 @@
         <dependency>
             <groupId>com.google.cloud</groupId>
             <artifactId>google-cloud-kms</artifactId>
+            <version>2.4.0</version>
```
Comment (exceptionfactory) [inline nifi-commons/nifi-property-protection-gcp/pom.xml:56]: This version change should be removed. The `google-cloud-kms` version is managed through the `libraries-bom` dependency.

File: nifi-registry/nifi-registry-core/pom.xml [hunk 2]
```diff
@@ -130,7 +130,7 @@
             <dependency>
                 <groupId>com.google.guava</groupId>
                 <artifactId>guava</artifactId>
-                <version>28.0-jre</version>
+                <version>31.0.1-jre</version>
```
Comment (exceptionfactory) [inline nifi-registry/nifi-registry-core/pom.xml:133]: The latest version of Guava is `31.1-jre`, which contains several updates to what is included in `31.0.1-jre`.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for contribution @mr1716! The Guava upgrades appear scoped to the NiFi Toolkit and NiFi Registry, and the successful automated workflow builds look good. However, since the latest version of Guava is `31.1-jre`, it would be better to move to that version instead of `31.0.1-jre`.

As noted in the details on `nifi-property-protection-gcp`, the dependency version for `google-cloud-kms` is managed through the `libraries-bom` dependency, so a specific version should not be specified in that location. Instead, the `libraries-bom` dependency version should be upgraded.

**Comment by mr1716** [comment]
@exceptionfactory Thanks for the feedback. So would this change upgrade the remaining use of Guava from 28 to the most recent version?

**Review by mr1716 — COMMENTED** [review]
Looks good. Unable to approve though

**Comment by mr1716** [comment]
@exceptionfactory Thanks for the information. I reviewed and approved, but is there anything I need to do? Should I submit a new PR instead?
 I also noticed that updating the Google-KMS software would resolve another JIRA ticket for Java Protobuf, as the newer version of Google KMS. Is there anything else I need to do for this?

**Comment by exceptionfactory** [comment]
> @exceptionfactory Thanks for the information. I reviewed and approved, but is there anything I need to do? Should I submit a new PR instead? I also noticed that updating the Google-KMS software would resolve another JIRA ticket for Java Protobuf, as the newer version of Google KMS. Is there anything else I need to do for this?

In light of the fact that upgrading Google KMS and other libraries impacts more than just Guava, it would be better to handle that in a separate Jira issue and separate PR.

If you can make the version adjustment to the Guava version in the other files, I can approve the workflow run.

**Comment by mr1716** [comment]
@exceptionfactory ok. I have removed the Google KMS change and kept the Guava. Is there anything else I need to do and was that not the right change?

**Comment by mr1716** [comment]
@exceptionfactory Yes sir. Once this is approved, I will either remove or update this fork accordingly.

**Review by exceptionfactory — APPROVED** [review]
Thanks for the upgrade @mr1716! The usage of Guava in NiFi Toolkit and Registry is minimal, and basic tests for both work as expected. +1 merging

**Pull request closed without merging**